swagger.yaml

```diff
@@ -5096,8 +5096,7 @@ definitions:
         description: Build ID of the RHCOS image.
       support_level:
         type: string
-        enum: [beta, production]
```

Comment:
If the point of the PR is to allow the UI to use the information, isn't turning this into a free-form string going against that goal?

Reply:
I agree. I would expect that we would have meaningful names that UI for example could use.
i.e.
* 4.8 - dev_preview
* 4.7 - production / GA / full_support (=default)
* 4.6 - maintenance_support
* 4.4 - EOL

Reply:
None of the values above are unique: Full support has overlapped period of 2 versions. I have added "Recommended" to the default version and this is the string that the UI will look for. With hard coded enum structure I will not be able to do that

Reply:
Unless we want to artificially mark the selected default version as "production" and demote artificially the previous version to "maintenance" However then it would not be align with the official life cycle as declared by https://access.redhat.com/support/policy/updates/openshift

Reply:
I would recommend leaving it as enum.  The UI can either have some heuristic for the recommended version (e.g., look for the latest version that is full_support), or we can have a separate field for that.

Reply:
Very well. I will change the enum description of the support_level to match the openshift terms and add is_default field for the use of the UI

Reply:
@YuviGold what was the original purpose of the support_level field? Maybe now it can be remove entirely if it is not use for UI presentation?

Reply:
The original purpose was to add extra data about the version which until now wasn't used.
I think that adding a new field only for "is_default" is redundant while it could be included in the support_level.

I don't think that we need to match the OCP life cycle, I suggested it as a source to decide how to categorize versions.

Reply:
@slaviered please take with the UI guys about this change before pushing it to verify that it's not breaking anything.